### PR TITLE
Image optimisation and progressive JPEG support

### LINKED ISSUE #10: Add possibility for image optimisation
Could piggyback off http://jpegclub.org/jpegtran/, http://pmt.sourceforge.net/pngcrush/ or similar libraries

### PR DESCRIPTION
**Pull request by kaedroho**
Fixes #10 

This PR implements image optimisation using Pillows ``optimize`` feature. It also implements a switch to convert JPEG files to progressive which should improve user experience when used for images on the web.

We may still want to support external image optimisers as I think they probably have a few more tricks than Pillow does, but this PR is definitely an improvement.

### CODE REVIEW

File: docs/guide/save.rst [hunk 1]
```diff
@@ -30,3 +30,28 @@ For example, to save an image with low quality:
 
     with open('low_quality.jpg', 'wb') as f:
         i.save_as_jpeg(f, quality=40)
+
+Progressive JPEGs
+-----------------
+
+:meth:`~Image.save_as_jpeg` takes a ``progressive`` keyword argument, that can
+be used to make it output a progressive JPEG file:
```
Comment (tomdyson) [inline docs/guide/save.rst]: Does it output progressive JPEGs by default?

File: docs/guide/save.rst [hunk 2]
```diff
@@ -30,3 +30,28 @@ For example, to save an image with low quality:
...
+Optimized images
+----------------
+
+:meth:`~Image.save_as_jpeg` and :meth:`~Image.save_as_png` both take an
+``optimise`` keyword that when set to true, will output an optimized image.
+
+.. code-block:: python
+
+    with open('optimized.jpg', 'wb') as f:
+        i.save_as_jpeg(f, optimize=True)
```
Comment (tomdyson) [inline docs/guide/save.rst]: is it 'optimise' or 'optimize'? You have 'ise' in the description and 'ize' in the example.

File: docs/guide/save.rst (hunk 1 above)
Comment (tomdyson) [inline docs/guide/save.rst]: Now I've seen that JPEGs are not progressive by default - we should clarify this in the docs.

File: docs/guide/save.rst [hunk 3]
```diff
@@ -30,3 +30,28 @@ For example, to save an image with low quality:
...
+.. code-block:: python
+
+    with open('progressive.jpg', 'wb') as f:
+        i.save_as_jpeg(f, progressive=True)
+
+Optimized images
+----------------
+
+:meth:`~Image.save_as_jpeg` and :meth:`~Image.save_as_png` both take an
+``optimise`` keyword that when set to true, will output an optimized image.
```
Comment (tomdyson) [inline docs/guide/save.rst]: We should clarify what optimisation does, and what type of images it works on (JPEGs and PNGs?).

**Comment by tomdyson** [comment]
Looks good, apart from my doc niggles!

File: docs/guide/save.rst (hunk 2 above)
Comment (kaedroho) [inline docs/guide/save.rst]: It should be 'ize' everywhere in the code but 'ise' when I discuss it in the PR. I know, confusing!

**Comment by kaedroho** [comment]
Thanks @tomdyson!

File: docs/guide/save.rst (hunk 2 above)
Comment (tomdyson) [inline docs/guide/save.rst]: It's currently 'ise' in save.rst, line 49.

**Comment by joshblum** [comment]
Any update for this getting merged?

**Pull request merged**